src/BenchmarkDotNet/Toolchains/InProcess/InProcessExecutor.cs

```diff
@@ -54,69 +65,97 @@ public InProcessExecutor(TimeSpan timeout, BenchmarkActionCodegen codegenMode, b
-            var oldThreadPriority = thread.Priority;
-
-            var affinity = parameters.BenchmarkCase.Job.ResolveValueAsNullable(EnvironmentMode.AffinityCharacteristic);
-            try
-            {
-                process.TrySetPriority(ProcessPriorityClass.High, parameters.Logger);
-                thread.TrySetPriority(ThreadPriority.Highest, parameters.Logger);
 
-                if (affinity != null)
+            if (Synchronous)
+            {
+                try
                 {
-                    process.TrySetAffinity(affinity.Value, parameters.Logger);
+                    exitCode = InProcessRunner.Run(host, parameters.BenchmarkCase, CodegenMode);
+                }
+                catch (Exception ex)
+                {
+                    parameters.Logger.WriteLineError($"// ! {GetType().Name}, exception: {ex}");
                 }
-
-                exitCode = InProcessRunner.Run(host, parameters.BenchmarkCase, CodegenMode);
-            }
-            catch (Exception ex)
-            {
-                parameters.Logger.WriteLineError($"// ! {GetType().Name}, exception: {ex}");
             }
-            finally
+            else
             {
-                process.TrySetPriority(oldPriority, parameters.Logger);
-                thread.TrySetPriority(oldThreadPriority, parameters.Logger);
+                var process = Process.GetCurrentProcess();
+                var oldPriority = process.PriorityClass;
+                var oldAffinity = process.TryGetAffinity();
+                var thread = Thread.CurrentThread;
+                var oldThreadPriority = thread.Priority;
+
+                var affinity = parameters.BenchmarkCase.Job.ResolveValueAsNullable(EnvironmentMode.AffinityCharacteristic);
+                try
+                {
+                    process.TrySetPriority(ProcessPriorityClass.High, parameters.Logger);
+                    thread.TrySetPriority(ThreadPriority.Highest, parameters.Logger);
 
-                if (affinity != null && oldAffinity != null)
+                    if (affinity != null)
+                    {
+                        process.TrySetAffinity(affinity.Value, parameters.Logger);
+                    }
+
+                    exitCode = InProcessRunner.Run(host, parameters.BenchmarkCase, CodegenMode);
+                }
+                catch (Exception ex)
                 {
-                    process.TrySetAffinity(oldAffinity.Value, parameters.Logger);
+                    parameters.Logger.WriteLineError($"// ! {GetType().Name}, exception: {ex}");
+                }
+                finally
+                {
+                    process.TrySetPriority(oldPriority, parameters.Logger);
+                    thread.TrySetPriority(oldThreadPriority, parameters.Logger);
+
+                    if (affinity != null && oldAffinity != null)
+                    {
+                        process.TrySetAffinity(oldAffinity.Value, parameters.Logger);
+                    }
```

Comment:
I think that it would be better to rename the existing `ExecuteCore` method to `ExecuteCoreAsync` and introduce a new method for synchronous run. This big `if/else` block is hard to read.